projects/aca-content/src/lib/components/dl-custom-components/datatable-cell-badges/datatable-cell-badges.component.spec.ts

```diff
@@ -112,4 +100,28 @@ describe('DatatableCellBadgesComponent', () => {
     const dynamicComponent = fixture.debugElement.query(By.css('adf-dynamic-component')).nativeElement;
     expect(dynamicComponent).toBeDefined();
   });
+
+  describe('mouse and keyboard events', () => {
+    let badges: HTMLElement[];
+    let runActionSpy: jasmine.Spy;
+    beforeEach(() => {
+      spyOn(appExtensionService, 'getBadges').and.returnValue(
+        of([{ id: 'test', type: ContentActionType.custom, icon: 'warning', tooltip: 'test tooltip', actions: { click: 'test' } }])
+      );
+      component.ngOnInit();
+      fixture.detectChanges();
+      badges = fixture.debugElement.queryAll(By.css('.adf-datatable-cell-badge')).map((badge) => badge.nativeElement);
```

Comment:
maybe use harness here too?

Reply:
With harness we can only test `mat-icon` name, tooltip is a property of `adf-icon`, that can't be tested with harness